rswag-specs/lib/rswag/specs/request_factory.rb

```diff
@@ -242,11 +242,13 @@ def add_payload(request, parameters, example)
         content_type = request[:headers]['CONTENT_TYPE']
         return if content_type.nil?
 
-        if ['application/x-www-form-urlencoded', 'multipart/form-data'].include?(content_type)
-          request[:payload] = build_form_payload(parameters, example)
-        else
-          request[:payload] = build_json_payload(parameters, example)
-        end
+        request[:payload] = if ['application/x-www-form-urlencoded', 'multipart/form-data'].include?(content_type)
+                              build_form_payload(parameters, example)
+                            elsif content_type == 'application/json'
```

Comment:
@romanblanco This check is too simplistic to consider all valid json types. For example, the JSON:API content type is `[application/vnd.api+json](http://www.iana.org/assignments/media-types/application/vnd.api+json)`